libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionColumnFactory.php

```diff
@@ -23,4 +32,32 @@ private function getBaseTypeFromMetadata(array $metadata): ?string
         }
         return null;
     }
+
+    private function getNativeDataType(array $metadata): ?DefinitionInterface
+    {
+        $type = null;
+        $options = [];
+        foreach ($metadata as $metadatum) {
+            switch ($metadatum['key']) {
+                case Common::KBC_METADATA_KEY_TYPE:
+                    $type = $metadatum['value'];
+                    break;
+                case Common::KBC_METADATA_KEY_LENGTH:
+                    $options['length'] = $metadatum['value'];
+                    break;
+                case Common::KBC_METADATA_KEY_DEFAULT:
+                    $options['default'] = $metadatum['value'];
+                    break;
+                case Common::KBC_METADATA_KEY_NULLABLE:
+                    $options['nullable'] = $metadatum['value'];
+                    break;
+                default:
+                    break;
+            }
+        }
+        if ($type && $this->nativeDatatypeClass) {
```

Comment:
no point in figuring out type when nativeDataTypeClasss is empty

Reply:
Can't we just obtain the `nativeDatatypeClass` from metadata here? So it doesn't have to be passed from TableDefinition?

Reply:
hmmm, the info comes from the table metadata rather than the column metadata, but yeah, can probably pass it here?  I'll try a couple variations